agent-inject/agent/config.go

```diff
@@ -8,6 +8,8 @@ import (
 
 const (
 	DefaultTemplate = "{{ with secret \"%s\" }}{{ range $k, $v := .Data }}{{ $k }}: {{ $v }}\n{{ end }}{{ end }}"
+	TokenTemplate   = `{{ with secret "auth/token/lookup-self" }}{{ .Data.id }}{{ end }}`
```

Comment:
One small request: add a newline to the end of the `.Data.id` line (like the default template) because if you were to look at it in the container:

```
/vault/secrets $ ls
ca.cert      db-creds     server.cert  server.key   token
/vault/secrets $ cat token
s.gXuAjfknTq0dK1i5s5rsjxpI/vault/secrets $
```

Reply:
Added in d9dcbcddb2d915921112e0e98258fec3052bce40

Reply:
Hmm, it's literally writing the `\n` characters, not a new line.

Reply:
Fixed in c2a7199e572fa02d3ea60c1154b188f5842cb3e6